hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -319,12 +319,13 @@ private HoodiePairData<String, HoodieRecord<HoodieMetadataPayload>> doLookup(Hoo
    */
   private HoodieData<HoodieRecord<HoodieMetadataPayload>> doLookupIndexRecords(HoodieData<String> keys, String partitionName, List<FileSlice> fileSlices,
                                                                                SerializableFunctionUnchecked<String, String> keyEncodingFn) {
+    boolean isSecondaryIndex = MetadataPartitionType.fromPartitionPath(partitionName).equals(MetadataPartitionType.SECONDARY_INDEX);
 
     final int numFileSlices = fileSlices.size();
     if (numFileSlices == 1) {
       List<String> keysList = keys.map(keyEncodingFn::apply).collectAsList();
       TreeSet<String> distinctSortedKeys = new TreeSet<>(keysList);
-      return lookupRecords(partitionName, new ArrayList<>(distinctSortedKeys), fileSlices.get(0));
+      return lookupRecords(partitionName, new ArrayList<>(distinctSortedKeys), fileSlices.get(0), !isSecondaryIndex);
```

Comment:
How are column stats records are looked up through these APIs?  For column stats index it can use prefix lookup, i.e., look up records by using column name only, whereas the column stats records use the keys constructed by column name, partition name, and file name.

Reply:
Got it.  After reading the code, column stats are read by using `HoodieBackedTableMetadata#getRecordsByKeyPrefixes` which calls `HoodieBackedTableMetadata#lookupRecords` directly, not going through `HoodieBackedTableMetadata#doLookupIndexRecords`.  The methods are bit confusing now in terms of the scope, i.e., which indexes use certain API.  We should do a follow-up refactoring.